game/ffmpeg.hh

```diff
@@ -146,15 +144,15 @@ class AudioBuffer {
 	}
 	bool wantMore() { return m_pos < m_posReq + m_data.capacity() / 2; }
 	/// Should the input stop waiting?
-	bool condition() { return m_quit || wantMore() || wantSeek(); }
+	bool condition() { return m_quit.load() || wantMore() || wantSeek(); }
 	mutable mutex m_mutex;
-	boost::condition m_cond;
+	std::condition_variable_any m_cond;
 	boost::circular_buffer<int16_t> m_data;
-	size_t m_pos;
-	int64_t m_posReq;
-	unsigned m_sps;
-	double m_duration;
-	bool m_quit;
+	size_t m_pos = 0;
+	int64_t m_posReq = 0;
+	unsigned m_sps = 0;
+	double m_duration = getNaN();
+	std::atomic<bool> m_quit;
```

Comment:
what was wrong with bool?

Reply:
Making it atomic avoids having to use mutex+lock when only the flag is used. The rest of Performous uses volatile bool m_quit for similar effect (written before c++11) but they should all be changed into std::atomic<bool>.